web/src/components/ClusterDistribution/ClusterDistributionPage.tsx

```diff
@@ -28,6 +29,8 @@ const CLUSTERS_STATE = CLUSTERS.reduce((result, cluster) => {
   return { ...result, [cluster]: { enabled: true } }
 }, {})
 
+const CLUSTER_BUILD_NAMES: Map<string, string> = new Map(getClusters().map((c) => [c.display_name, c.build_name]))
+
```

Comment:
Instead of this, could we pass the whole `cluster` object from parent (and grandparent, grandgrandparent,... if necessary), and use `cluster.build_name` and `cluster.build_name`? I'm afraid current code might trigger unnecessary re-renders.

Reply:
This is tricky due to filtering logic in the component, so let's keep it this way for now and address it Later™ :)